resolver/addressable_resolver.go

```diff
@@ -258,6 +261,11 @@ func (r *URIResolver) addressableFromDestinationRef(ctx context.Context, dest du
 		addr.CACerts = dest.CACerts
 	}
 
+	if dest.Audience != nil && *dest.Audience != "" {
+		// destinations audience takes preference
+		addr.Audience = dest.Audience
+	}
```

Comment:
Shouldn't these 2 logic should be similar but they aren't? as in they both prioritize `dest` or they prioritize `addr`

Reply:
The TLS one seems wrong to me

Reply:
https://github.com/knative/pkg/issues/2837